src/Stratis.Bitcoin.Features.SmartContracts/SmartContractPosPowBlockDefinition.cs

```diff
@@ -0,0 +1,166 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using Microsoft.Extensions.Logging;
+using NBitcoin;
+using Stratis.Bitcoin.Consensus;
+using Stratis.Bitcoin.Features.Consensus;
+using Stratis.Bitcoin.Features.Consensus.CoinViews;
+using Stratis.Bitcoin.Features.Consensus.Interfaces;
+using Stratis.Bitcoin.Features.MemoryPool;
+using Stratis.Bitcoin.Features.MemoryPool.Interfaces;
+using Stratis.Bitcoin.Features.Miner;
+using Stratis.Bitcoin.Features.SmartContracts.Consensus;
+using Stratis.Bitcoin.Mining;
+using Stratis.Bitcoin.Utilities;
+using Stratis.SmartContracts.Core;
+using Stratis.SmartContracts.Core.State;
+using Stratis.SmartContracts.Core.Util;
+
+namespace Stratis.Bitcoin.Features.SmartContracts
+{
+    /// <summary>
+    /// Defines how a proof of work block will be built on a proof of stake network.
+    /// </summary>
+    public sealed class SmartContractPosPowBlockDefinition : BlockDefinition
```

Comment:
I assume this is the same approach as Stratis? First N blocks PoW, then switch to PoS?

For the sidechain we won't quite need that.

Reply:
Yeah this is more so that we can mine on a POS network ;)